### ISSUE TITLE
startProcessInstanceWithForm

### ISSUE THREAD

**Issue by mzhifu** [flowable/flowable-engine#1907]
**Describe the bug**
when i start process by this way :  runtimeService.startProcessInstanceWithForm();
because when repositoryService deploy process and formRepositoryService deploy form i was set a tenant_id, next i run command startProcessInstanceWithForm,always error a not 
 have formDefinition and formKey, but in my database this formKey was exist.
so, i have question, the tenant_id is a single people tenant_id, or tenant_id is a single enterprise.
or this is a  bug????
**Expected behavior**
start a process and form 
**Code**
runtimeService.startProcessInstanceWithForm();
**Additional context**
use flowable 6.4.2
Springboot 2.1.1.RELEASE
ORM :  mybatis-plus
database: pgsql

**Comment by unamanic** [comment]
The best place to ask questions is at https://forum.flowable.org/, you are more likely to get an answer and that answer will be searchable by others with the same issue. 

I think your issue is this:
- You have deployed a process/form with a tenant id
- You have attempted to start the process without a tenant id
- The process does not start.

When you start a process without specifying a tenant id, it uses the default tenant.  the default does not have access to any processes (or forms, etc) in the other tenants.  When you start a process using a tenant it will have access to anything deployed to that tenant and in the default tenant.  How you use tenants is up to you (per person, per business unit, per customer, etc).  

Bottom line is that if you deploy a process with a tenant id, you should also start that process with a tenant id.

**Comment by mzhifu** [comment]
sorry, the forum in my country always do not open, or the fornum server too long distance my country, the page always was loading, so  i m very dejected ask for in github issue.
go back the question, startProcessInstanceWithForm  , this method  can't found start by a tenant id, only have some start by message, but i have to start with form . so finally i only createFormInstance  api  create a form instance.  thank you reply.

**Comment by tijsrademakers** [comment]
Hi,

When you look at the implementation of the startProcessInstanceWithForm method then you can see that it's using the ProcessInstanceBuilder:

https://github.com/flowable/flowable-engine/blob/master/modules/flowable-engine/src/main/java/org/flowable/engine/impl/RuntimeServiceImpl.java#L161

So instead of using this method you can use RuntimeService createProcessInstanceBuilder and then do the same as in the startProcessInstanceWithForm method implementation and add the tenantId to it as well.

**Comment by mzhifu** [comment]
O(∩_∩)O~  thank you very much

**Comment by izerui** [comment]
good
